deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/objdetect/Yolo2OutputLayer.java

```diff
@@ -661,10 +668,10 @@ public void clearNoiseWeightParams() {
                         px += x;
                         py += y;
 
-                        if(sm == null){
-                            try (MemoryWorkspace wsO = Nd4j.getMemoryManager().scopeOutOfWorkspaces()) {
-                                sm = softmax.get(point(i), all(), point(x), point(y)).dup();
-                            }
+
+                        INDArray sm;
+                        try (MemoryWorkspace wsO = Nd4j.getMemoryManager().scopeOutOfWorkspaces()) {
+                            sm = softmax.get(point(i), point(box), all(), point(x), point(y)).dup();
```

Comment:
I believe x and y are reversed here? Would explain why we had trouble with the classification part... @yquemener

Reply:
Most definitely!